src/internal/connector/graph_connector.go

```diff
@@ -108,6 +116,59 @@ func NewGraphConnector(
 	return &gc, nil
 }
 
+// PopulateOwnerIDAndNamesFrom takes the provided owner identifier and produces
+// the owner's name and ID from that value.  Returns an error if the owner is
+// not recognized by the current tenant.
+//
+// The id-name maps are optional.  Some processes will look up all owners in
+// the tenant before reaching this step.  In that case, the data gets handed
+// down for this func to consume instead of performing further queries.  The
+// maps get stored inside the gc instance for later re-use.
+//
+// TODO: If the maps are nil or empty, this func will perform a lookup on the given
+// owner, and populate each map with that owner's id and name for downstream
+// guarantees about that data being present.  Optional performance enhancement
+// idea: downstream from here, we should _only_ need the given user's id and name,
+// and could store minimal map copies with that info instead of the whole tenant.
+func (gc *GraphConnector) PopulateOwnerIDAndNamesFrom(
+	owner string, // input value, can be either id or name
+	ins common.IDNameSwapper,
+) (string, string, error) {
+	// move this to GC method
+	id, name, err := getOwnerIDAndNameFrom(owner, ins)
+	if err != nil {
+		return "", "", errors.Wrap(err, "resolving resource owner details")
+	}
+
+	gc.IDNameLookup = ins
+
+	if len(ins.IDs()) == 0 && len(ins.Names()) == 0 {
+		gc.IDNameLookup = common.IDsNames{
+			IDToName: map[string]string{id: name},
+			NameToID: map[string]string{name: id},
+		}
+	}
```

Comment:
either `ins` will be nil and needs created or it will be set to some value by earlier code
```suggestion
	if ins == nil {
		gc.IDNameLookup = common.NewIDNameSwapper()
		return "", "", nil
	}
	
	// move this to GC method
	id, name, err := getOwnerIDAndNameFrom(owner, ins)
	if err != nil {
		return "", "", errors.Wrap(err, "resolving resource owner details")
	}
	
	gc.IDNameLookup = ins
```